astropy/units/tests/test_units.py

```diff
@@ -503,6 +503,13 @@ def test_pickling():
     assert other is u.m
 
     new_unit = u.IrreducibleUnit(['foo'], format={'baz': 'bar'})
+    # Test pickling unregistered unit
+    p = pickle.dumps(new_unit)
+    new_unit_copy = pickle.loads(p)
+    assert new_unit_copy.names == ['foo']
+    assert new_unit_copy.get_format_name('baz') == 'bar'
+
+    # Test pickling registered unit
     with u.add_enabled_units([new_unit]):
         p = pickle.dumps(new_unit)
         new_unit_copy = pickle.loads(p)
```

Comment:
This test doesn't quite seem to test that if `load` the unit, it automatically becomes registered. But not quite part of your PR, of course...